### ISSUE TITLE
IDE0032 should not be reported for volatile fields

### ISSUE THREAD

**Issue by azyobuzin** [dotnet/roslyn#25379]
**Version Used**: Visual Studio Community 15.6.1

**Steps to Reproduce**:
Here is the example code.

```csharp
class Class1
{
    private volatile int _field1;

    public int Property1
    {
        get { return this._field1; }
        set { this._field1 = value; }
    }

    public int Property2 { get; set; }
}
```

`Property1` is a wrapper for `_field1`, which has `volatile` modifier. `Property2` is a normal auto property, which is written for comparison.

Visual Studio says `_field1` can be an auto property.

![Screenshot of Code Action](https://user-images.githubusercontent.com/1000655/37224196-1675352a-2416-11e8-93ef-1df266f81806.png)

**Expected Behavior**: This suggestion should not be shown because the field has `volatile` modifier.

The way to access to volatile fields is different from the way to access to normal fields. Access to volatile fields needs `volatile.` prefix in IL. So the field cannot be an auto property in the present case.

**Comment by Neme12** [comment]
PR here: #25431
